### ISSUE TITLE
Where are the sources ??

### ISSUE THREAD

**Issue by rogerclarkmelbourne** [esp8266/esptool#1]
Where are the sources ??

I'd like to recompile the WIndows exe to try some baud rates compatible with the Silicon Labs CP2102 which are currently being rejected

**Comment by 0x2b3bfa0** [comment]
They seem to be missing. However you can try to use this Python version of `esptool`: https://github.com/themadinventor/esptool

**Comment by crushedice2000** [comment]
They seem to be missing. However you can try to use this Python version of `esptool`: https://github.com/themadinventor/esptool

**Comment by rogerclarkmelbourne** [comment]
Yes. I know I can use the Python version.

However, this is still an issue that needs to be resolved, as Open Source projects should strive to include sources for all the binaries that are compiled as part of the project

**Comment by 0x2b3bfa0** [comment]
Yes, this is a <s>good</s> important issue. However, the programmer's procrastination seems to be great... Let's wait until he takes a look here. **;-)**

**Comment by 0x2b3bfa0** [comment]
I'm not a Windows expert (I don't used it later than when I was 9 years old). However, I've made a bash script for *nix users to use as wrapper. It allows me to use the `esptool` binary to create the firmware and esptool.py to upload it:

```
#!/bin/bash

case "$1" in
  ("-eo")
    ./esptool.bin $@;;
  ("-cd")
    ./esptool.py --port "$6" write_flash 0x00000 "${10}";;
  (*) echo "Invalid build command!"
esac
```

You should put this script in `/path/to/Arduino15/packages/esp8266/tools/esptool/0.4.5/esptool` and give execute permissions.

**Comment by rogerclarkmelbourne** [comment]
@crushedice2000

Its not a big issue for me at the moment, as I can use the esptool binary at 115200, I just wanted to change it, as it has a list of "supported" data rates, which doesn't totally match with the CP2101 based USB to Serial board I'm using - at least not for the higher speeds

I can modify things to use python if I really need to .

**Comment by 0x2b3bfa0** [comment]
Ok. Let's see what says the maintainer. :smile:

**Comment by 0x2b3bfa0** [comment]
@rogerclarkmelbourne: I've discovered the project source code: https://github.com/igrr/esptool-ck

**Comment by 0x2b3bfa0** [comment]
I've made a pull request
